src/app/plugins/githubuploadadapter.js

```diff
@@ -103,8 +103,32 @@ export class Adapter {
 						// Step 2: the real upload takes place this time to Amazon S3 servers,
 						// using information returned from Step 1.
 
-						// The final URL of the file is already known, even before the upload. We save it here.
-						returnUrl = response.asset.href;
+						const assetUrl = typeof response.asset_upload_url === 'string' ? response.asset_upload_url : null;
```

Comment:
We need a comment in the code explaining why is this done in a first place. Otherwise we won't understand this in a month from now.

Reply:
It makes a perfect sense to wrap this new code into a helper function with a good name, like `sendAssetRequest` or `requestAssetConfirmation`.